pay/contracts/v1_upfront_pay/HectorPay.sol

```diff
@@ -423,14 +414,50 @@ contract HectorPay is
         }
     }
 
+    function activeStreamsByRemoveEnded(
+        address from
+    ) external returns (uint256 count) {
+        uint256 length = activeStreams[from].length();
+
+        for (uint256 i = 0; i < length; i++) {
+            bytes32 streamId = activeStreams[from].at(count);
+            Stream memory stream = streams[streamId];
+
+            if (stream.ends < block.timestamp || stream.lastPaid == 0) {
+                activeStreams[from].remove(streamId);
+                delete streams[streamId];
+            } else {
+                unchecked {
+                    ++count;
+                }
+            }
+        }
+    }
+
+    function isAvailableForActiveNewStream(
```

Comment:
@bejaxer I think we need to outsource the "logic" to check for things like limitationOfActiveStreams to another contract because since business is not finalized on the "conditions" to determine "when" an account is not able to stream when it hits xyz condition. 
- As a result, having the HectorPay contract assume this responsibility is not a good design. 
- So we'll create a new smart contract called SubscriptionValidation with IValidation interface
  - What it does: validate condition from provided inputs
  - Inputs: 
      - From: address
  - Output: boolean
- Then inside the HectorPay contract, we just need to reference to SubscriptionValidation contract.